### ISSUE TITLE
Support unknown inputs when using input files

### ISSUE THREAD

**Issue by dcreey** [abcxyz/abc#259]
### TL;DR

https://github.com/abcxyz/abc/issues/49 added support for input files via the `-input-file` flag. Unfortunately this requires that every spec have every input defined. We should support ignoring input values that are not defined in the spec.

### Detailed design

This has the unintended consequence of exposing unnecessary inputs/values to templates that don't need them.

Use case:
I have several templates all rendered using a centralized config file(s). I do not want to define input variables that are not used in the template files. Instead I would like to have the option to ignore input values from the input file that aren't defined in the spec.

**Comment by drevell** [comment]
FR accepted, either I or someone else will do this soon.

**Comment by drevell** [comment]
@stytchiz volunteered to take this one.

**Comment by stytchiz** [comment]
> Instead I would like to have the option to ignore input values from the input file that aren't defined in the spec.
Just to clarify, ignoring inputs in input files that are not defined in the spec would be default behavior after merging the above PR fix.
